### PR TITLE
Feature navigation

### PR DESCRIPTION
**Pull request by AdityaKhatri**

### CODE REVIEW

**Comment by eoglethorpe** [comment]
image

File: frontend/src/Root.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import React from 'react';
-import { BrowserRouter } from 'react-router-dom';
+import { Router } from 'react-router-dom';
 import { Provider } from 'react-redux';
 import { persistStore } from 'redux-persist';
+import browserHistory from './common/utils/routerUtils';
```
Comment (tnagorra) [inline frontend/src/Root.js]: ./common/utils/routerUtils should be named ./common/browserHistory

File: frontend/src/common/components/LinkOutsideRouter/index.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import CSSModules from 'react-css-modules';
+import PropTypes from 'prop-types';
+import React from 'react';
+import browserHistory from '../../utils/routerUtils';
+
+import styles from './styles.scss';
+
+const propTypes = {
```
Comment (tnagorra) [inline frontend/src/common/components/LinkOutsideRouter/index.js]: sort attributes please

File: frontend/src/common/components/LinkOutsideRouter/index.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    className: PropTypes.string,
+
+    children: PropTypes.oneOfType([
+        PropTypes.arrayOf(PropTypes.node),
+        PropTypes.node,
+    ]).isRequired,
+};
+
+const defaultProps = {
+    iconName: '',
```
Comment (tnagorra) [inline frontend/src/common/components/LinkOutsideRouter/index.js]: sort attributes please

File: frontend/src/common/components/LinkOutsideRouter/index.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    className: '',
+};
+
+@CSSModules(styles, { allowMultiple: true })
+export default class LinkOutsideRouter extends React.PureComponent {
+    static propTypes = propTypes;
+    static defaultProps = defaultProps;
+
+    onClick = (e) => {
+        e.preventDefault();
```
Comment (tnagorra) [inline frontend/src/common/components/LinkOutsideRouter/index.js]: should add some comments here

File: frontend/src/common/components/LinkOutsideRouter/index.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+        document.body.click();
+    }
+
+    render() {
+        return (
+            <a
+                href={this.props.to}
+                onClick={this.onClick}
+                className={this.props.className}
```
Comment (tnagorra) [inline frontend/src/common/components/LinkOutsideRouter/index.js:43]: Instead of using className, styles can be overridden from parent using "styles" prop

Comment (AdityaKhatri) [reply frontend/src/common/components/LinkOutsideRouter/index.js:43]: We tried that, but the underlying styles would be overridden too, so we went with this.

File: frontend/src/Root.js (hunk 1 above)
Comment (AdityaKhatri) [reply frontend/src/Root.js]: We might have to add other things on this. Are you suggesting we should rename it to browserHistory now, and if we add some other objects to this then change it to routerUtils?

Comment (tnagorra) [reply frontend/src/Root.js]: routerUtils in this case has created a global object "browserHistory", so the above naming would be more suitable
You can always use non-default exports for other objects that will be added.

File: frontend/src/common/components/LinkOutsideRouter/index.js (hunk 5 above)
Comment (tnagorra) [reply frontend/src/common/components/LinkOutsideRouter/index.js:43]: So this is the new way to override styles? I don't like this CSSModules.

Comment (AdityaKhatri) [reply frontend/src/common/components/LinkOutsideRouter/index.js:43]: I think both can be done, depending on the case

File: frontend/src/App.js [hunk 6]
```diff
@@ -89,6 +92,16 @@ export default class App extends React.Component {
                 component: () => <h1>{pageTitles.editEntries}</h1>,
                 private: true,
             },
+            {
+                path: '/users/:userId',
```
Comment (tnagorra) [inline frontend/src/App.js:96]: Y no substitue? :userId

Comment (AdityaKhatri) [reply frontend/src/App.js:96]: I think doing that after we write selectors for getting user data would be better.

**Pull request merged**